components/renderers/jsonforms-renderers.spec.ts

```diff
@@ -0,0 +1,242 @@
+            "label": {
+                "text": "LabeL"
+            },
+            "scope": {
+                "$ref": "#/properties/name"
+            }
+        };
+        scope.data = { "name": "My Name" };
+        let el = $compile('<jsonforms schema="schema" ui-schema="uiSchema" data="data"/>')(scope);
+        scope.$digest();
+        expect(el.find("label").text()).toEqual("LabeL");
+    }));
+
+    it("should render the provided label object with text attribute and show attribute set to true", inject(($rootScope, $compile) => {
+        let scope = $rootScope.$new();
+        scope.schema = {
+            "properties": {
+                "name": {
+                    "type": "string"
+                }
+            }
+        };
+        scope.uiSchema = {
+            "type": "Control",
+            "label": {
+                "text": "LabeL",
+                "show": true
+            },
+            "scope": {
+                "$ref": "#/properties/name"
+            }
+        };
+        scope.data = { "name": "My Name" };
+        let el = $compile('<jsonforms schema="schema" ui-schema="uiSchema" data="data"/>')(scope);
+        scope.$digest();
+        expect(el.find("label").text()).toEqual("LabeL");
+    }));
+
+    it("should hide the label when show attribute is set to false", inject(($rootScope, $compile) => {
+        let scope = $rootScope.$new();
+        scope.schema = {
+            "properties": {
+                "name": {
+                    "type": "string"
+                }
+            }
+        };
+        scope.uiSchema = {
+            "type": "Control",
+            "label": {
+                "show": false
+            },
+            "scope": {
+                "$ref": "#/properties/name"
+            }
+        };
+        scope.data = { "name": "My Name" };
+        let el = $compile('<jsonforms schema="schema" ui-schema="uiSchema" data="data"/>')(scope);
+        scope.$digest();
+        expect(el.find("label").text()).toBeFalsy();
+    }));
+
+    it("should render the default label when show attribute is set to true", inject(($rootScope, $compile) => {
+        let scope = $rootScope.$new();
+        scope.schema = {
+            "properties": {
+                "name": {
+                    "type": "string"
+                }
+            }
+        };
+        scope.uiSchema = {
+            "type": "Control",
+            "label": {
+                "show": true
+            },
+            "scope": {
+                "$ref": "#/properties/name"
+            }
+        };
+        scope.data = { "name": "My Name" };
+        let el = $compile('<jsonforms schema="schema" ui-schema="uiSchema" data="data"/>')(scope);
+        scope.$digest();
+        expect(el.find("label").text()).toEqual("Name");
+    }));
+
+    it("should hide the label when text is defined but show is set to false", inject(($rootScope, $compile) => {
+        let scope = $rootScope.$new();
+        scope.schema = {
```

Comment:
The schema seems to be the same in all cases, maybe we could extract it and declare it as a global var within the test